odd-collector-sdk/odd_collector_sdk/secrets/aws/ssm_parameter_store.py

```diff
@@ -0,0 +1,104 @@
+import boto3
+from yaml import safe_load
+
+from odd_collector_sdk.logger import logger
+from odd_collector_sdk.secrets.base_secrets import BaseSecretsBackend
+
+
+class SSMParameterStoreSecretsBackend(BaseSecretsBackend):
+    def __init__(self, **kwargs) -> None:
+        super().__init__()
+        self.region_name = kwargs.get("region_name", "us-east-1")
+        self.collector_id = kwargs.get("collector_id", "")
+        self.config_prefix = kwargs.get("config_prefix", "/odd/collector_config")
+        self.collector_settings_section_prefix = kwargs.get(
+            "collector_settings_section_prefix", "/collector_settings"
+        )
+        self.plugins_section_prefix = kwargs.get("plugins_section_prefix", "/plugins")
+        self.ssm_client = boto3.client("ssm", region_name=self.region_name)
```

Comment:
As I can see we are using `@property` decorator for public access to some class attributes. So if we are going to follow this approach we should show that these attributes are kind of private or protected and mark them accordingly.
for example:
```
self.__region_name = kwargs.get("region_name", "us-east-1") # private
self._region_name = kwargs.get("region_name", "us-east-1")  # protected
```

Reply:
@AndreyNenashev actually only methods get_collector_settings() and get_plugins() were planned to be accessible, so I renamed all other naming with _, 'marking' them private